components/google_analytics/google_analytics.app.mjs

```diff
@@ -1,11 +1,53 @@
+import { google } from "googleapis";
+
 export default {
   type: "app",
   app: "google_analytics",
-  propDefinitions: {},
+  propDefinitions: {
+    google,
+    token: {
+      type: "string",
+      label: "Authentication Token",
+      description: "Authentication Token",
+      optional: true,
+    },
+    accountId: {
+      type: "string",
+      label: "Account ID",
+      description: "Account ID to retrieve goals for",
+    },
+    webPropertyId: {
+      type: "string",
+      label: "Web Property ID",
+      description: "Web property ID to retrieve goals for",
+    },
+    profileId: {
+      type: "string",
+      label: "Profile ID",
+      description: "View (Profile) ID to retrieve goals for",
+    },
```

Comment:
As mentioned above, `accountId`, `webPropertyId`, and `profileId` should use async options if possible.